_data/compiler-options.yml

```diff
@@ -0,0 +1,1011 @@
+    - option: -Xmain-class
+      schema:
+        type: String
+        arg: path
+      description: "Class for manifest's Main-Class entry (only useful with `-d <jar>`)"
+    - option: -Xshow-class
+      schema:
+        type: String
+        arg: class
+      description: "Show internal representation of class."
+    - option: -Xshow-object
+      schema:
+        type: String
+        arg: object
+      description: "Show internal representation of object."
+    - option: -Xshow-phases
+      schema:
+        type: Boolean 
+      description: "Print a synopsis of compiler phases."
+    - option: -Xsource-reader
+      schema:
+        type: String
+        arg: classname
+      description: "Specify a custom method for reading source files."
+    - option: -Xreporter
+      schema:
+        type: String
+        arg: classname
+        default: scala.tools.nsc.reporters.ConsoleReporter
+      description: "Specify a custom reporter for compiler messages."
+    - option: -Xstrict-inference
+      schema:
+        type: Boolean 
+      description: "Don't infer known-unsound types."
+    - option: -Xsource
+      schema:
+        type: Scala Version
+        arg: version
+        default: 2.13
+      description: "Treat compiler input as Scala source for the specified version, see [scala/bug#8126](https://github.com/scala/bug/issues/8126)."
+    - option: -Xno-patmat-analysis
+      schema:
+        type: Boolean 
+      description: "Don't perform exhaustivity/unreachability analysis. Also, ignore `@switch` annotation."
+    - option: -Xfull-lubs
+      schema:
+        type: Boolean 
+      description: "Retains pre 2.10 behavior of less aggressive truncation of least upper bounds."
+    - option: -Xmixin-force-forwarders
+      schema:
+        type: Choice
+        arg: mode
+        default: true
+      description: "Generate forwarder methods in classes inhering concrete methods from traits."
+      choices:
+        - choice: true
+          description: "Always generate mixin forwarders."
+        - choice: junit
+          description: "Generate mixin forwarders for JUnit-annotated methods (JUnit 4 does not support default methods)."
+        - choice: false
+          description: "Only generate mixin forwarders required for program correctness."
+    - option: -Xxml
+      schema:
+        type: Choice
+        arg: property
+      description: "Configure XML parsing."
+      choices:
+        - choice: coalescing
+          description: "Convert PCData to Text and coalesce sibling nodes."
+
+
+
+- category: Private Settings
+  description: |
+    Options with `-Y` prefix are more experimental and unstable than those with `-X` prefix.
+
+    Those influencing to language semantics are likely to be deprecated in Scala 2.13 (see [scala/scala-dev#430](https://github.com/scala/scala-dev/issues/430)).
+
+    Some optimization-related options, e.g. `-opt:**` and `-opt-inline-form` are included in this section, since **REASON_HERE**.
```

Comment:
I found  `-opt:**` and `-opt-inline-form` are in private settings (-ish) section for `-Y' in [the source](https://github.com/scala/scala/blob/1c56f0af6d3d59b7d2a8dbcf64077b0c1fe90f07/src/compiler/scala/tools/nsc/settings/ScalaSettings.scala#L297) therefore put them in this section. 

Is it OK to categorize `-opt:**` and `-opt-inline-form` as private settings?

Reply:
yes I think so :smile:

Reply:
I think that should say `-opt-inline-from`, not `-form`

Reply:
Added reason and fixed typo.